lib/src/text_codec.dart

```diff
@@ -248,11 +248,16 @@ class PostgresTextDecoder<T extends Object> {
         // TODO: should we check for other representations (e.g. `1`, `on`, `y`,
         // and `yes`)?
         return (asText == 't' || asText == 'true') as T;
+      
+      case PgDataType.voidType:
+        // TODO: is returning `null` here is the appripriate thing to do? 
+        return null;
 
       // We could list out all cases, but it's about 20 lines of code.
+      // TODO: implement the rest of the types 
       // ignore: no_default_cases
       default:
-        throw UnimplementedError('Text decoding for $_dataType');
+        return asText as T;
```

Comment:
Maybe?
```
if (asText is T) return asText;
throw UnimplementedError('Text decoding for $_dataType');
```

Reply:
```dart
        if (asText is T) {
          return asText as T; // i had to do it like this; otherwise it gives an error 
        }
        throw UnimplementedError('Text decoding for $_dataType');
```